### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by iTrooz**
Anchor checking seem to not work very well in sphinx (see https://github.com/sphinx-doc/sphinx/issues/9016) so I disabled them altogether, and fixed the few other links that made the CI fail

The CI now indicate a pretty green :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    =
+SPHINXOPTS    = -D linkcheck_anchors=
```
Comment (TheAssassin) [inline Makefile:5]: Please add a comment that indicates why we do this (maybe with a link to this PR).

File: source/packaging-guide/hosted-services/travis-ci.rst [hunk 2]
```diff
@@ -52,7 +52,7 @@ Super simple uploading of continuous builds (each push) to GitHub Releases. If t
 Using ``upload.sh``
 ^^^^^^^^^^^^^^^^^^^
 
-The :code:`upload.sh` script in the `uploadtool repository <uploadtool-github>`__ is designed to be called from Travis CI after a successful build. By default, this script will *delete* any pre-existing release tagged with :code:`continuous`, tag the current state with the name :code:`continuous`, create a new release with that name, and upload the specified binaries there. For pull requests, it will upload the binaries to transfer.sh instead and post the resulting download URL to the pull request page on GitHub.
+The :code:`upload.sh` script in the `uploadtool repository <https://github.com/probonopd/uploadtool>`__ is designed to be called from Travis CI after a successful build. By default, this script will *delete* any pre-existing release tagged with :code:`continuous`, tag the current state with the name :code:`continuous`, create a new release with that name, and upload the specified binaries there. For pull requests, it will upload the binaries to transfer.sh instead and post the resulting download URL to the pull request page on GitHub.
```
Comment (TheAssassin) [inline source/packaging-guide/hosted-services/travis-ci.rst:55]: The link is defined in the line below. If it is not working, please fix it there. If there is a reason this is not possible, please tell us why.

**Review by TheAssassin — CHANGES_REQUESTED** [review]
Thanks! This is a long-awaited PR, I guess. It's very much reasonable to just disable a non-working check for now. Checking whether the pages exist at all is already useful. I suppose Sphinx can't handle, e.g., dynamically composed websites, only static content. Please see my comments.

File: source/packaging-guide/hosted-services/travis-ci.rst (hunk 2 above)
Comment (iTrooz) [reply source/packaging-guide/hosted-services/travis-ci.rst]: I don't think the `_uploadtool-github` "variable" worked at all before.
I updated the syntax to follow the one on https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages

**Comment by iTrooz** [comment]
All done !

**Pull request merged**